src/request.js

```diff
@@ -63,18 +63,25 @@ export function expandPathWithKeyValue(path, key, value, options) {
 
 
 /* Expand paths with variable substitutions.
+   Remove used vairable from args
  */
 export function expandPath(context, path, args) {
     const options = get(context, 'options', {});
     let expandedPath = path;
-    Object.keys(args || {}).forEach((key) => {
+    const keys = Object.keys(args || {});
+    keys.forEach((key) => {
         const value = args[key];
         if (typeof (value) === 'string') {
             // URI encode the value; handling embedded `/`
             const expandedValue = value.split('/').map(
                 part => encodeURIComponent(part),
             ).join('/');
-            expandedPath = expandPathWithKeyValue(expandedPath, key, expandedValue, options);
+            const newPath = expandPathWithKeyValue(expandedPath, key, expandedValue, options);
```

Comment:
@pelegdvir I think worth adding a one-line comment explaining what this logic achieves as non-obvious if we look back at this a month from now.